core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java

```diff
@@ -107,104 +100,112 @@ protected final Logger logger() {
+        }
+        return requestLogLevelMapper.orElse(RequestLogLevelMapper.of(LogLevel.DEBUG));
     }
 
     /**
-     * Sets the {@link LogLevel} to use when logging failure responses (e.g., failed with an exception).
-     * If unset, will use {@link LogLevel#WARN}. The request will be logged too if it was not otherwise.
+     * Sets the {@link LogLevel} to use when logging responses that's status is equal to the specified
+     * {@link HttpStatus}.
      */
-    public LoggingDecoratorBuilder failureResponseLogLevel(LogLevel failedResponseLogLevel) {
-        if (isResponseLogLevelMapperSet) {
-            throw new IllegalStateException("responseLogLevelMapper has been set already.");
-        }
-        this.failedResponseLogLevel = requireNonNull(failedResponseLogLevel, "failedResponseLogLevel");
-        isResponseLogLevelSet = true;
-        return this;
+    public LoggingDecoratorBuilder responseLogLevel(HttpStatus status, LogLevel logLevel) {
+        return responseLogLevelMapper(ResponseLogLevelMapper.of(status, logLevel));
     }
 
     /**
-     * Returns the {@link LogLevel} to use when logging failure responses (e.g., failed with an exception).
+     * Sets the {@link LogLevel} to use when logging responses that's status is belong to the specified
+     * {@link HttpStatusClass}.
      */
-    @VisibleForTesting
-    final LogLevel failedResponseLogLevel() {
-        return failedResponseLogLevel;
+    public LoggingDecoratorBuilder responseLogLevel(HttpStatusClass statusClass, LogLevel logLevel) {
+        return responseLogLevelMapper(ResponseLogLevelMapper.of(statusClass, logLevel));
     }
 
     /**
-     * Sets the {@link Function} to use when mapping the log level of request logs.
+     * Sets the {@link LogLevel} to use when logging successful responses (e.g., no unhandled exception).
+     * If unset, will use {@link LogLevel#DEBUG}.
      */
-    public LoggingDecoratorBuilder requestLogLevelMapper(
-            Function<? super RequestOnlyLog, LogLevel> requestLogLevelMapper) {
-        if (isRequestLogLevelSet) {
-            throw new IllegalStateException("requestLogLevel has been set already.");
-        }
-        this.requestLogLevelMapper = requireNonNull(requestLogLevelMapper, "requestLogLevelMapper");
-        isRequestLogLevelMapperSet = true;
-        return this;
+    public LoggingDecoratorBuilder successfulResponseLogLevel(LogLevel successfulResponseLogLevel) {
+        requireNonNull(successfulResponseLogLevel, "successfulResponseLogLevel");
+        return responseLogLevelMapper(log -> log.responseCause() == null ? successfulResponseLogLevel : null);
     }
 
     /**
-     * Returns the {@link LogLevel} to use when logging request logs.
+     * Sets the {@link LogLevel} to use when logging failure responses (e.g., failed with an exception).
+     * If unset, will use {@link LogLevel#WARN}. The request will be logged too if it was not otherwise.
```

Comment:
> The request will be logged too if it was not otherwise.

I wasn't able to catch how this was directly related to this api 😅 

```suggestion
     * {@link LogLevel#WARN} will be used by default.
```